twake/frontend/src/app/scenes/Client/ChannelsBar/Modals/WorkspaceChannelList.tsx

```diff
@@ -24,36 +37,29 @@ export default () => {
           suffix={
             <Icon type="search" className="m-icon-small" style={{ color: 'var(--grey-dark)' }} />
           }
-          autoFocus
           placeholder={Languages.t('scenes.client.channelbar.workspacechannellist.autocomplete')}
-          onChange={(e: React.ChangeEvent<HTMLInputElement>) => console.log(channels)}
+          value={search}
+          onChange={(event: React.ChangeEvent<HTMLInputElement>) => setSearch(event.target.value)}
+          autoFocus
         />
       </Row>
-
       <PerfectScrollbar
         style={{ maxHeight: '250px' }}
         component="div"
         options={{ suppressScrollX: true, suppressScrollY: false }}
       >
-        {channels &&
-          channels.map((channel: ChannelResource, index: number) => {
+        {autoChannels
+          .filter(({ name }) => name.toUpperCase().indexOf(search.toUpperCase()) > -1)
+          .map((channel, index) => {
             return (
-              <div key={`${channel.data.id}_${index}`}>
+              <div key={`${channel.id}_${index}`}>
                 <WorkspaceChannelRow channel={channel} />
                 <Divider style={{ margin: 0 }} />
               </div>
             );
           })}
-        {channels.length === loadedChannels && (
-          <Row align="middle" justify="center" className="y-margin">
-            <Typography.Link
-              style={{ color: 'var(--grey-dark)' }}
-              onClick={() => setLoadedChannels(loadedChannels + 3)}
-            >
-              {Languages.t('scenes.client.channelbar.channelmemberslist.loader')}
-            </Typography.Link>
-          </Row>
-        )}
+        {/* {!autoChannels.filter(({ name }) => name.toUpperCase().indexOf(search.toUpperCase()) > -1)
+          .length && setLimit(autoChannels.length + 2)} */}
```

Comment:
Try something like this to not call the function setLimit on each render:
`setLimit(autoChannels.length + 2)` => `limit < autoChannels.length + 2 && setLimit(autoChannels.length + 2)`

Reply:
Done :+1: